source/common/network/listener_impl.cc

```diff
@@ -14,36 +14,34 @@
 
 namespace Network {
 
-void ListenerImpl::listenCallback(evconnlistener*, evutil_socket_t fd, sockaddr* addr, int,
+void ListenerImpl::listenCallback(evconnlistener*, evutil_socket_t fd, sockaddr* remote_addr, int,
                                   void* arg) {
   ListenerImpl* listener = static_cast<ListenerImpl*>(arg);
 
-  if (listener->use_original_dst_) {
-    sockaddr_storage orig_dst_addr;
-    memset(&orig_dst_addr, 0, sizeof(orig_dst_addr));
+  sockaddr_storage local_addr;
+  memset(&local_addr, 0, sizeof(local_addr));
 
-    bool success = Utility::getOriginalDst(fd, &orig_dst_addr);
+  bool success = Utility::getOriginalDst(fd, &local_addr);
```

Comment:
If the connection was not redirected, getOriginalDst will return the destination address that the client had specified.
The call will return false only when the getsockopt call itself fails.
That will only happen if the socket is not valid, not accepted or in some other error condition.